uint8 channel
uint16 pwm
uint32 speed
uint32 time